exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -220,7 +225,21 @@ public QProfiles getProfilesJSON() {
 
       final List<ProfileInfo> finishedQueries = Lists.newArrayList();
 
-      final Iterator<Map.Entry<String, QueryProfile>> range = completed.getRange(0, MAX_PROFILES);
+      //Defining #Profiles to load
+      int maxProfilesToLoad;
+      if (MaxProfiles == null) {
+        MaxProfiles = work.getContext().getConfig().getInt(ExecConstants.HTTP_MAX_PROFILES);
+      }
+
+      String maxProfilesParams = uriInfo.getQueryParameters().getFirst(MAX_QPROFILES_PARAM);
+      if (maxProfilesParams != null && !maxProfilesParams.isEmpty()) {
+        maxProfilesToLoad = Integer.valueOf(maxProfilesParams);
+      } else {
+        maxProfilesToLoad = MaxProfiles;
+      }
```

Comment:
Agreed. Got rid of the MaxProfiles altogether. We'll read the config value and load the appropriate value accordingly.